src/json.ts

```diff
@@ -0,0 +1,33 @@
+import fs, { PathLike } from 'fs';
+
+export const SPACES_TO_INDENT_JSON = 2;
+
+export function parseJsonFile(jsonFile: PathLike, allowMissing: boolean = true): object {
+    if (allowMissing && !fs.existsSync(jsonFile)) {
+        return {};
+    }
+
+    let parsedJsonFile;
+
+    try {
+        parsedJsonFile = JSON.parse(fs.readFileSync(jsonFile).toString());
+    } catch (error) {
+        let failedReason = JSON.stringify(error);
+
+        if (error instanceof SyntaxError) {
+            failedReason = `${error.message}`;
+        }
+
+        throw new Error(`Failed to parse ${jsonFile}: ${failedReason}`);
+    }
```

Comment:
Wouldn't it suffice for the json exception to bubble up?

Reply:
This file here is re-used and thus, used for multiple JSON files. Thats why I wanted to add the JSON file path to the error here.